File: Selenium/Activities/Activity18.java
package activities;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import io.github.bonigarcia.wdm.WebDriverManager;

import java.util.List;

public class Activity18 {
    public static void main(String[] args){
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Title is :" + driver.getTitle());
        WebElement select_element = driver.findElement(By.id("multi-select"));
        Select multi_dropdown = new Select(select_element);
        multi_dropdown.selectByVisibleText("Javascript");
        System.out.println("First selected option is :" + multi_dropdown.getFirstSelectedOption().getText());
        multi_dropdown.selectByIndex(4);
        multi_dropdown.selectByIndex(5);
        multi_dropdown.selectByIndex(6);
        multi_dropdown.selectByValue("node");
        List<WebElement> lst1 = multi_dropdown.getAllSelectedOptions();
        System.out.println("All Options Selected :");
        for (WebElement l1 : lst1){
            System.out.println(l1.getText());
        }
        multi_dropdown.deselectByIndex(5);
        List<WebElement> lst2 = multi_dropdown.getAllSelectedOptions();
        System.out.println("All Options Selected post deselect:");
        for (WebElement l2 : lst2){
            System.out.println(l2.getText());
        }
        driver.close();
    }
}

File: Selenium/Activities/Activity18.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/selects")
    print("Title is :",driver.title)
    select_element = driver.find_element(By.ID,"multi-select")
    multi_drpdwn = Select(select_element)
    multi_drpdwn.select_by_visible_text("Javascript")
    print("First selected Option is :",multi_drpdwn.first_selected_option.text)
    for i in range(4,7):
        multi_drpdwn.select_by_index(i)
    multi_drpdwn.select_by_value("node")
    lst1 = multi_drpdwn.all_selected_options
    print("All the selected options :")
    for l1 in lst1:
        print(l1.text)
    multi_drpdwn.deselect_by_index(5)
    lst2 = multi_drpdwn.all_selected_options
    print("All the selected options post deselect:")
    for l2 in lst2:
        print(l2.text)
    driver.close()

